File: core/src/main/java/com/lomekwi/cave/pipeline/image/Transformable.java
package com.lomekwi.cave.pipeline.image;

public interface Transformable extends Renderable {
    Transform getTransform();
    void setTransform(Transform transform);
    float getBaseWidth();
    float getBaseHeight();
    default void reset() {
        Transform t = getTransform();
        t.reset(0, 0, getBaseWidth(), getBaseHeight());
    }
}

File: core/src/main/scala/com/lomekwi/cave/pipeline/image/Transformable.scala
package com.lomekwi.cave.pipeline.image

trait Transformable extends Renderable {
  def getTransform(): Transform
  def setTransform(transform: Transform): Unit
  def getBaseWidth(): Float
  def getBaseHeight(): Float
  def reset(): Unit = {
    val t = getTransform()
    t.reset(0, 0)
  }
}
